### PR TITLE
fixes for geometry less layers

### PR DESCRIPTION
**Pull request by 3nids**
depends on https://github.com/qgis/QGIS/pull/8667

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-413-0c914009cf8883597cba0dccd51db2ca05d46c0b-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-413-0c914009cf8883597cba0dccd51db2ca05d46c0b-x86.apk)

File: src/core/qgismobileapp.cpp [hunk 1]
```diff
@@ -139,7 +139,7 @@ void QgisMobileapp::initDeclarative()
   qRegisterMetaType<QgsPoint>( "QgsPoint" );
   qRegisterMetaType<QgsPointXY>( "QgsPointXY" );
   qRegisterMetaType<QgsCoordinateTransformContext>( "QgsCoordinateTransformContext" );
-  qRegisterMetaType<QgsWkbTypes::GeometryType>( "QgsWkbTypes::GeometryType" );
+  qRegisterMetaType<QgsWkbTypes::GeometryType>( "QgsWkbTypes::GeometryType" ); // could be removed since we have now qmlRegisterUncreatableType<QgsWkbTypes> ?
```
Comment (3nids) [inline src/core/qgismobileapp.cpp]: ping @m-kuhn

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-413-36995aef5b20edb6045acbaaeb6d2427b335dc85-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-413-36995aef5b20edb6045acbaaeb6d2427b335dc85-x86.apk)

File: src/core/qgismobileapp.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgismobileapp.cpp]: I guess so

**Pull request merged**